pkg/parcacol/querier.go

```diff
@@ -169,7 +219,18 @@ func (q *Querier) Values(
 }
 
 func MatcherToBooleanExpression(matcher *labels.Matcher) (logicalplan.Expr, error) {
-	ref := logicalplan.Col("labels." + matcher.Name)
+	labelExpr, err := matcherToBinaryExpression(matcher, logicalplan.Col(ColumnLabelsPrefix+matcher.Name))
+	if err != nil {
+		return nil, err
+	}
+	pprofLabelExpr, err := matcherToBinaryExpression(matcher, logicalplan.Col(ColumnPprofLabelsPrefix+matcher.Name))
+	if err != nil {
+		return nil, err
+	}
+	return logicalplan.Or(labelExpr, pprofLabelExpr), nil
```

Comment:
Like I said on Discord, I'm not sure this is quite correct, we actually need to say something like "label.a exists and is equal to "b" and pprof_labels.a does not exist OR pprof.labels_a exists and is equal to "b" and labels.a does not exist"

Unfortunately, frostdb doesn't support this yet.

I think we can leave this part for last though, we might have it implemented by the time we fix all the other things! :)